datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -58,21 +59,42 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
     @PostMapping("/bookstores/book")
     protected CollectionModel<EntityModel<Book>> oneBookToManyStores(@RequestBody Book book, @RequestParam List<String> id) throws Exception {
         List<EntityModel<Book>> entityList = new ArrayList<>();
-        for(String storeId: id)
-        {
-            book.setStoreID(Long.parseLong(storeId));
-            if(!this.map.containsKey(Long.parseLong(storeId))) continue;
-            HttpURLConnection con = createConnection(this.map.get(Long.parseLong(storeId)) + "/books", "POST");
+        if (!this.storeRepository.findAll().get(0).getServerId().equals(this.leader.getLeader())) {
+            String address = this.map.get(this.leader.getLeader());
+            address = address.substring(0,address.lastIndexOf("/") + 1) + "book?id=" + id.toString().replaceAll("[\\[ \\]]", "");
             Gson gson = new Gson();
-            JsonObject jso = book.makeJson();
-            outputJson(con, gson, jso);
-            entityList.add(assembler.toModel(new Book(book)));
+            String json = gson.toJson(book.makeJson());
+            HttpResponse<String> response = getPostConnectionJava9(address, json);
```

Comment:
what happens if the response is not success here?

Reply:
The Hub is constantly checking on the leader. IDK how it can possibly not connect

Reply:
haha. never assume success.... :)